src/client/components/2dContainer/blueprint.controller.js

```diff
@@ -20,38 +21,492 @@ export default class BlueprintEditor {
+                    {}, {},
+                ],
+            },
         };
-        Object.keys(Maths).forEach((key) => {
-            Object.keys(copy).forEach((key1) => {
-                if (Math.abs(Maths[key] - copy[key1]) > maxDiff) {
-                    maxDiff = Math.abs(Maths[key] - copy[key1]);
-                    console.log(maxDiff);
-                    if (Maths[key] < Maths[key1]) {
-                        vals.minVal = key;
-                        vals.maxVal = key1;
+
+        // Find the location of each frame and insert their offsets into the structure
+        let count = 0;
+        let startY;
+        boat.frames.forEach((frame, index) => {
+            const {locationA, locationB, locationC} = findLocation(boat, frame);
+            location1 = locationA;
+            location2 = locationB;
+            location3 = locationC;
+            //  console.log(location1, location2, location3);
+            Object.keys(frames).forEach((key) => {
+                if (frames[key].count === i) {
+                    // const prevY = currY;
+                    if (frames[key].color !== 'invisible') {
+                        frames[key].points[0].x = 15;
+                        startY = currY + 15;
+                        frames[key].points[0].y = startY;
+                        frames[key].points[1].x = Math.abs(location1.x - location2.x) + frames[key].points[0].x;
+                        frames[key].points[1].y = Math.abs(location1.y - location2.y) + frames[key].points[0].y;
+                        frames[key].points[2].x = Math.abs(location2.x - location3.x) + frames[key].points[1].x;
+                        frames[key].points[2].y = Math.abs(location2.y - location3.y) + frames[key].points[1].y;
+                        frames[key].points[3].x = Math.abs(location2.x - location3.x) + frames[key].points[2].x;
+                        frames[key].points[3].y = Math.abs(Math.abs(location2.y - location3.y) - frames[key].points[2].y);
+                        frames[key].points[4].x = Math.abs(location1.x - location2.x) + frames[key].points[3].x;
+                        frames[key].points[4].y = Math.abs(Math.abs(location1.y - location2.y) - frames[key].points[3].y);
+                        currY = frames[key].points[2].y;
+                        frames[key].size = this.pythagorean(frames[key].points[2].x, frames[key].points[1].x, frames[key].points[2].y, frames[key].points[1].y) * 2;
                     } else {
-                        vals.minVal = key1;
-                        vals.maxVal = key;
+                        if (frames[key].variable === 'frameSide') {
+                            frames[key].points[0].x = 16;
+                            frames[key].points[0].y = startY;
+                            frames[key].points[1].x = Math.abs(location1.x - location2.x) + frames[key].points[0].x;
+                            frames[key].points[1].y = Math.abs(location1.y - location2.y) + frames[key].points[0].y;
```

Comment:
This is the only place that I could find you using the location variables. This means that you actually don't even need to declare them at all. What you're currently doing is equivalent to:
let i = 0;
let j = i;
//do something with j.
Instead, just use locationA, B, C directly.